sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1926,6 +1926,13 @@ class DatasetSuite extends QueryTest
       }
     }
   }
+
+  test("SPARK-32585: Support scala enumeration in ScalaReflection") {
+    checkDataset(
+      Seq(FooClassWithEnum(1, FooEnum.E1), FooClassWithEnum(2, FooEnum.E2)).toDS(),
+      Seq(FooClassWithEnum(1, FooEnum.E1), FooClassWithEnum(2, FooEnum.E2)): _*
```

Comment:
Related to the above comment, I would add a test case where the enum value is `null`.

Reply:
Added the null test.